modules/articles/server/controllers/articles.server.controller.js

```diff
@@ -81,7 +81,29 @@ exports.delete = function (req, res) {
  * List of Articles
  */
 exports.list = function (req, res) {
-  Article.find().sort('-created').populate('user', 'displayName').exec(function (err, articles) {
+  // Prepare paging parameters
+  var page = Math.max(0, req.query.page -1); // using a zero-based page index for use with skip()
+  var take = req.query.take || 1000;
+
+  /** 
+  * I don't like having the filter, so I'm removing it for now. It's going to introduce a bit of complexity. 
+  * For instance, when we're querying the collection statistics, to get the total count used with the paging.
+  * We may have discrepencies between the total amount of documents in the collection, and what this filter will return.
+  * An alternative option would be to return the count of documents returned from this query/filter. However, this would
+  * require us to query first for the count & then again using our limit() & skip() conditions.
+  *
+  * Something in the form of:
+  *  { articles: [], count: x }
+  *
+  * The idea behind the filter is that mongoose will start at the beginning of the collection, and move it's way up to 
+  * satisfy the skip condition. This could have performance issues. ref: http://stackoverflow.com/questions/5539955/how-to-paginate-with-mongoose-in-node-js/23640287#23640287
+  */
+
+  //var today = new Date(Date.now());
+  //var dateFloor = (new Date()).setDate(today.getDate() - 365);
+  //var filter = { created: { $gte: req.query.createdAfter || dateFloor } };
+
+  Article.find().limit(take).skip(page * take).sort('-created').populate('user', 'displayName').exec(function (err, articles) {
```

Comment:
I sort of jumped the gun on submitting this PR. I mostly wanted to get it out there to gain some momentum, and see what we could do without an outside dependency. If we can determine how we want this feature to work by keeping it basic, then we'll be in a much better position to choose the best library (if at all).